frontend/src/pages/QueryBuilderChooser/QueryBuilderChooser.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { FC, useState, useEffect } from 'react';
+import * as localForage from 'localforage';
+import { localForageKeys, api } from '../../utils';
+import { Alert, Button, Modal } from 'react-bootstrap';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import { ICheckData, IRadio } from '../../components/IRadio';
+import { CheckBox } from '../../components/CheckBox';
+import axios from 'axios';
+
+type SelectedQueryBuilder = 'basic' | 'advanced';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const QueryBuilderChooser: FC<RouteComponentProps> = () => {
+  const [showModal, setShowModal] = useState(true);
+  const [selectedOption, setSelectedOption] = useState<SelectedQueryBuilder>('advanced');
+  const [checked, setChecked] = useState(false);
+  const [choice, setChoice] = useState<string>();
+
+  const toggleModal = () => {
+    setShowModal(!showModal);
+    history.push('/queries/build/advanced/');
+  };
+  const onRadioChange = (data: ICheckData) => {
+    setSelectedOption(data.value as SelectedQueryBuilder);
+  };
+
+  const history = useHistory();
+  useEffect(() => {
+    localForage.getItem<string>(localForageKeys.PREFERENCES).then((key) => {
+      if (key) {
+        setChoice(key || undefined);
+        if (key === 'advanced') {
+          history.push('/queries/build/advanced/');
+        } else {
+          history.push('/queries/build/basic/');
+        }
+      }
+    });
+  }, []);
+
+  const handleChange = () => {
+    setChecked(!checked);
+  };
+
+  const handleSave = () => {
+    localForage.getItem<string>(localForageKeys.API_KEY).then((token) => {
+      const userPreference = {
+        preferences: selectedOption,
+        global_choice: false,
+      };
+      axios
+        .request({
+          method: 'post',
+          url: api.routes.USERPREFERENCE,
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `token ${token}`,
+          },
+          data: userPreference,
+        })
+        .then(() => {
+          if (selectedOption || choice) {
+            if (selectedOption === 'basic' || choice === 'basic') {
```

Comment:
Reason why i have both is so that someone can still be redirected to the querybuilder without the querybuilder of their choice being saved. Then should they select the remember choice checkbox, their choice will be saved and also they will be redirected. However, Let me do away with one so that they have no option but to select the remember choice checkbox.

Reply:
Nope... they don't need to select. You can do this with only one of them without changing the existing behaviour. Also, if a change I've suggested requires you to change the behaviour, DON'T do it till you check with me.

Reply:
Noted. I will have some that are doing so. I will check in with you in the morning then